tidy3d/components/medium.py

```diff
@@ -663,8 +696,36 @@ def _passivity_validation(cls, val, values):
         if not values.get("allow_gain") and val < 0:
             raise ValidationError(
                 "For passive medium, 'conductivity' must be non-negative. "
-                "To simulate gain medium, please set 'allow_gain=True'. "
-                "Caution: simulations with gain medium are unstable, and are likely to diverge."
+                "To simulate a gain medium, please set 'allow_gain=True'. "
+                "Caution: simulations with a gain medium are unstable, and are likely to diverge."
+            )
+        return val
+
+    @pd.validator("permittivity", always=True)
+    def _permittivity_modulation_validation(cls, val, values):
+        """Assert modulated permittivity cannot be <= 0."""
```

Comment:
I assume your to-do item regarding fixing the CFL number is for the cases when 0 < permittivity < 1. Practically, this sounds like some very esoteric material, I'm almost tempted to say we just validate it out right now, but it may be of interest to e.g. some academic users, so if it's not too hard to do the CFL modification then I guess it would be good.

Reply:
In lots of cases, it's probably 0<permittivity<1 when a dispersive medium of eps_inf=1 is modulated. But the modulation is in general very weak in EO, so the modulated permittivity is very close to one. It's easy to fix CFL number.

Reply:
Added now, together with a bunch of validators.